packages/nuxt/src/app/composables/fetch.ts

```diff
@@ -205,3 +205,21 @@ export function useLazyFetch<
   // @ts-expect-error we pass an extra argument with the resolved auto-key to prevent another from being injected
   autoKey)
 }
+
+function generateOptionSegments <_ResT, DataT, DefaultT>(opts: UseFetchOptions<_ResT, DataT, any, DefaultT, any, any>) {
+  const segments: Array<string | undefined | Record<string, string>> = [
+    toValue(opts.method as MaybeRef<string | undefined> | undefined)?.toUpperCase() || 'GET',
+    toValue(opts.baseURL),
+  ]
+  for (const _obj of [opts.params || opts.query, opts.headers]) {
+    const obj = toValue(_obj)
+    if (obj) {
+      const unwrapped: Record<string, string> = {}
+      for (const key in obj) {
+        unwrapped[key] = toValue(obj[key as keyof typeof obj])
```

Comment:
Have you checked iteration node worngly goes through objects/class instances with a custom constructor such as `Headers`?

Reply:
Nope, good point 👍